theano/sandbox/cuda/opt_util.py

```diff
@@ -100,9 +101,7 @@ def opt(node):
                     # other cases are too complex for now
                     return None
                 if W.broadcastable != targ.inputs[out_in].broadcastable:
-                    # Would need to explicitly tile the output to fill
-                    # the full shape here.  Disable for now.
-                    return None
+                    W = broadcast_like(W, targ.inputs[out_in], node.fgraph)
```

Comment:
I think the opposite. So time it before we merge this. The elemwise will probably be inplace and have scalar being a special case that is optimized.